parsl/monitoring/db_logger.py

```diff
@@ -57,24 +121,40 @@ def get_db_logger(host='search-parsl-logging-test-2yjkk2wuoxukk2wdpiicl7mcrm.us-
     OptionalModuleMissing
 
     """
-    logger = logging.getLogger(__file__)
-    if enable_es_logging:
+    logger = logging.getLogger(logger_name)
+    if db_logger_config_object is None:
+        logger.addHandler(NullHandler())
+        return logger
+
+    if db_logger_config_object.logger_type == 'elasticsearch':
         if not _es_logging_enabled:
             raise OptionalModuleMissing(
                 ['CMRESHandler'], "Logging to ElasticSearch requires the cmreslogging module")
 
-        handler = CMRESHandler(hosts=[{'host': host,
-                                       'port': port}],
-                               use_ssl=True,
+        handler = CMRESHandler(hosts=[{'host': db_logger_config_object.host,
+                                       'port': db_logger_config_object.port}],
+                               use_ssl=db_logger_config_object.enable_ssl,
                                auth_type=CMRESHandler.AuthType.NO_AUTH,
-                               es_index_name=index_name,
+                               es_index_name=db_logger_config_object.index_name,
                                es_additional_fields={
                                    'Campaign': "test",
-                                   # use the name of the user's home directory as their username since there
-                                   # does not seem to be a portable way to do this
-                                   'Version': version,
+                                   'Version': db_logger_config_object.version,
                                    'Username': getpass.getuser()})
-        logger = logging.getLogger("ParslElasticsearch")
+        logger = logging.getLogger(db_logger_config_object.logger_name)
+        logger.setLevel(logging.INFO)
+        logger.addHandler(handler)
+    elif db_logger_config_object.logger_type == 'local_database' and not is_logging_server:
+        # add a handler that will pass logs to the logging server
+        handler = RemoteHandler(db_logger_config_object.web_app_host, db_logger_config_object.web_app_port)
+        # use the specific name generated by the server or the monitor wrapper
+        logger = logging.getLogger(logger_name)
+        logger.setLevel(logging.INFO)
+        logger.addHandler(handler)
+    elif db_logger_config_object.logger_type == 'local_database' and is_logging_server:
+        # add a handler that will take logs being recieved on the server and log them to the database
+        handler = DatabaseHandler(db_logger_config_object.eng_link)
+        # use the specific name generated by the server or the monitor wrapper
+        logger = logging.getLogger(logger_name)
         logger.setLevel(logging.INFO)
         logger.addHandler(handler)
     else:
```

Comment:
you should never hit this else, right? because you've gone over all the valid cases of logger_type? (if so, then throw an exception indicating that the code is broken if you hit it rather than silently failing)